iac-extensions/docker/src/main/java/org/sonar/iac/docker/tree/impl/DockerImageImpl.java

```diff
@@ -59,6 +60,14 @@ public List<Instruction> instructions() {
     return instructions;
   }
 
+  @Override
+  public boolean isLastDockerImageInFile() {
+    Body parent = (Body) parent();
+    List<DockerImage> dockerImages = parent.dockerImages();
+    DockerImage last = dockerImages.get(dockerImages.size() - 1);
+    return last == this;
+  }
+
```

Comment:
Not quite sure if this logic is in the right place here. I think I would have put it in `TreeUtils`. Especially with the call to `parent()`

Reply:
I disagree. The `*Utils` classes are a sign of a bad design, not Object Oriented Programming.
Our `DockerTree` implementations are in most cases data structures, not rich objects. Our `DockerTree`s look like anemic entities in a typical project with Hibarnate. 

Here is a simple logic, only a real-life instance of `DockerImageImpl` is needed.
It's only valid for `DockerImageImpl`. The `TreeUtils` contains valid functions for all kinds of `Tree` implementations. (BTW IMO they should be moved to the `Tree` interface).